src/hooks/useAuth.ts

```diff
@@ -10,42 +10,65 @@ import {
   WalletConnectConnector,
 } from '@web3-react/walletconnect-connector';
 import toast from 'components/Basic/Toast';
-import { connectorLocalStorageKey } from '../config';
+import { LS_KEY_IS_USER_LOGGED_IN } from '../config';
 import { connectorsByName, ConnectorNames } from '../utilities/connectors';
 import { setupNetwork } from '../utilities/wallet';
 
 const useAuth = () => {
   const { activate, deactivate } = useWeb3React();
   const login = useCallback(
-    (connectorID: ConnectorNames) => {
+    async (connectorID: ConnectorNames) => {
       const connector = connectorsByName[connectorID];
-      if (connector) {
-        activate(connector, async error => {
-          if (error instanceof UnsupportedChainIdError) {
-            const hasSetup = await setupNetwork();
-            if (hasSetup) {
-              activate(connector);
-            }
-          } else {
-            window.localStorage.removeItem(connectorLocalStorageKey);
-            if (error instanceof NoEthereumProviderError || error instanceof NoBscProviderError) {
-              toast.error({ title: 'No provider was found' });
-            } else if (
-              error instanceof UserRejectedRequestErrorInjected ||
-              error instanceof UserRejectedRequestErrorWalletConnect
-            ) {
-              if (connector instanceof WalletConnectConnector) {
-                const walletConnector = connector;
-                walletConnector.walletConnectProvider = undefined;
-              }
-              toast.error({ title: 'Please authorize to access your account' });
-            } else {
-              toast.error({ title: error.message });
-            }
-          }
+      if (!connector) {
+        toast.error({
+          title: 'An internal error occurred: wrong connector config. Please try again later',
```

Comment:
Would this happen if someone the user tries to connect with an unsupported wallet? I think that would be a nicer error message for the user and should also be good information to debug if it ever happened

Reply:
I think that's the idea yeah, I'm not too sure to be honest. When I saw that, I actually thought to myself that this case should never happen as we are the ones defining what wallets can be connected or not, so the only way this issue would come up is if an incorrect `connectorID` is passed to the login function.

I agree we can use a nicer message then, maybe something like "An internal error occurred: unsupported wallet. Please try again later". The reason I'd still keep the "internal error" part in the message is because as I mentioned I think this case should never happen if our typing is correct.

Reply:
yeah which would make it an error to Sentry in addition to a nice user error message